src/identity/actions/thunks/index.ts

```diff
@@ -0,0 +1,107 @@
+import {GetState} from 'src/types'
+import {Dispatch} from 'react'
+
+// Actions
+import {
+  setQuartzMe,
+  setQuartzMeStatus,
+  Actions as MeActions,
+} from 'src/me/actions/creators'
+import {
+  setCurrentOrgDetails,
+  setCurrentBillingProvider,
+  setQuartzIdentity,
+  setQuartzIdentityStatus,
+  Actions as IdentityActions,
+} from 'src/identity/actions/creators'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+// Utilities
+import {
+  fetchQuartzIdentity,
+  fetchAccountDetails,
+  fetchOrgDetails,
+} from 'src/identity/apis/auth'
+import {convertIdentityToMe} from 'src/identity/utils/convertIdentityToMe'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+// Thunks
+import {getQuartzMeThunk} from 'src/me/actions/thunks'
+
+export const getQuartzIdentityThunk = () => async dispatch => {
+  if (!isFlagEnabled('quartzIdentity')) {
+    dispatch(getQuartzMeThunk())
+    return
+  }
+
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const quartzIdentity = await fetchQuartzIdentity()
+
+    dispatch(setQuartzIdentity(quartzIdentity))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const legacyMe = convertIdentityToMe(quartzIdentity)
+    dispatch(setQuartzMe(legacyMe, RemoteDataState.Done))
+    dispatch(setQuartzMeStatus(RemoteDataState.Done))
+  } catch (error) {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Error))
+    dispatch(setQuartzMeStatus(RemoteDataState.Error))
+    console.error(error)
```

Comment:
Right now, when quartz (quartz me or identity) is pinged by the UI, the app doesn't make use of `RemoteDataState.Error` or do anything with the error (other than logging it). The app can't meaningfully carry on in a cloud environment if it can't fetch the user's information from quartz.

An improvement here (and probably elsewhere in the UI) would be to convert these `console.log` or `console.error` calls to toasts that notify the user there's an auth issue.